analyzers/src/SonarAnalyzer.CSharp/Rules/ClearTextProtocolsAreSensitive.cs

```diff
@@ -0,0 +1,165 @@
+            });
+
+        private void VisitObjectCreation(SyntaxNodeAnalysisContext context)
+        {
+            var objectCreation = (ObjectCreationExpressionSyntax)context.Node;
+
+            if (telnetRegexForIdentifier.IsMatch(objectCreation.Type.ToString()))
+            {
+                context.ReportDiagnosticWhenActive(Diagnostic.Create(DefaultRule, objectCreation.GetLocation(), TelnetKey, recommendedProtocols[TelnetKey]));
+            }
+            else if (!IsServerSafe(objectCreation) && objectInitializationTracker.ShouldBeReported(objectCreation, context.SemanticModel))
+            {
+                context.ReportDiagnosticWhenActive(Diagnostic.Create(EnableSslRule, objectCreation.GetLocation()));
+            }
+        }
+
+        private void VisitInvocationExpression(SyntaxNodeAnalysisContext context)
+        {
+            var invocation = (InvocationExpressionSyntax)context.Node;
+            if (telnetRegexForIdentifier.IsMatch(invocation.Expression.ToString()))
+            {
+                context.ReportDiagnosticWhenActive(Diagnostic.Create(DefaultRule, invocation.GetLocation(), TelnetKey, recommendedProtocols[TelnetKey]));
+            }
+        }
+
+        private void VisitAssignments(SyntaxNodeAnalysisContext context)
+        {
+            var assignment = (AssignmentExpressionSyntax)context.Node;
+            if (assignment.Left is MemberAccessExpressionSyntax memberAccess
+                && memberAccess.IsMemberAccessOnKnownType(EnableSslName, KnownType.System_Net_FtpWebRequest, context.SemanticModel)
+                && context.SemanticModel.GetConstantValue(assignment.Right) is { HasValue: true, Value: bool enableSslValue }
+                && !enableSslValue)
+            {
+                context.ReportDiagnosticWhenActive(Diagnostic.Create(EnableSslRule, assignment.GetLocation()));
+            }
+        }
+
+        private void VisitStringExpressions(SyntaxNodeAnalysisContext c)
+        {
+            var text = GetText(c.Node);
+            if (GetUnsafeProtocol(text) is {} unsafeProtocol)
+            {
+                c.ReportDiagnosticWhenActive(Diagnostic.Create(DefaultRule, c.Node.GetLocation(), unsafeProtocol, recommendedProtocols[unsafeProtocol]));
+            }
+        }
+
+        private bool IsServerSafe(ObjectCreationExpressionSyntax objectCreation) =>
+            objectCreation.ArgumentList?.Arguments.Count > 0
+            && validServerRegex.IsMatch(GetText(objectCreation.ArgumentList.Arguments[0].Expression));
+
+        private string GetUnsafeProtocol(string text)
+        {
+            if (httpRegex.IsMatch(text))
+            {
+                return "http";
+            }
+            else if (ftpRegex.IsMatch(text))
+            {
+                return "ftp";
+            }
+            else if (telnetRegex.IsMatch(text))
+            {
+                return "telnet";
+            }
+
+            return null;
```

Comment:
According to convention, to keep the flows explicitly trivial. VB.NET has better syntax for this :trollface: 😄 
```suggestion
        else
        {
            return null;
        }
```